ubuntu_image/common_builder.py

```diff
@@ -211,6 +211,27 @@ def pre_populate_bootfs_contents(self):
                 os.makedirs(target_dir, exist_ok=True)
         self._next.append(self.populate_bootfs_contents)
 
+    @classmethod
+    def _selective_copytree(cls, src, dst):
+        # Selectively copy entries from src to dst directory. Entries already
+        # existing at dst are not overwritten when overwrite is True.
+        os.makedirs(dst, exist_ok=True)
+        shutil.copystat(src, dst)
+        for entry in os.scandir(src):
+            sname = os.path.join(src, entry.name)
+            dname = os.path.join(dst, entry.name)
+
+            if entry.is_file() or entry.is_symlink():
+                if os.path.exists(dname):
+                    # skip existing entries when asked to do so
+                    continue
+                # copy files and recreate symlinks
+                shutil.copy(sname, dname)
```

Comment:
Reading the copytree() documentation, it seems that it uses copy2() by default instead of copy(), as it also copies over all the metadata. I think we should also use copy2() here to stay consistent. Not sure if that was the intention, as we used copy() everywhere else - of well!

Reply:
Also, another thing worth thinking about here: previously in such cases we've been calling copytree(symlinks=True). From what I see copy() (and copy2()) by default has follow_symlinks=True, which does the opposite, e.g. does not preserve symlinks but copies their contents instead.
So I think we might need to do a copy(..., follow_symlinks=False) here?